pyo3-macros-backend/src/method.rs

```diff
@@ -742,3 +722,29 @@ const IMPL_TRAIT_ERR: &str = "Python functions cannot have `impl Trait` argument
 const RECEIVER_BY_VALUE_ERR: &str =
     "Python objects are shared, so 'self' cannot be moved out of the Python interpreter.
 Try `&self`, `&mut self, `slf: PyRef<'_, Self>` or `slf: PyRefMut<'_, Self>`.";
+
+fn ensure_signatures_on_valid_method(
+    fn_type: &FnType,
+    signature: Option<&SignatureAttribute>,
+    text_signature: Option<&TextSignatureAttribute>,
+) -> syn::Result<()> {
+    if let Some(signature) = signature {
+        if let FnType::Getter(_) | FnType::Setter(_) | FnType::ClassAttribute = fn_type {
+            bail_spanned!(signature.kw.span() => "signature not allowed with this method type")
```

Comment:
Since the span is on the signature and not the getter/setter/class attribute, I think it would be helpful to call out the method type in question.